autoload/direnv.vim

```diff
@@ -0,0 +1,83 @@
+" direnv.vim - support for direnv <http://direnv.net>
+" Author:       zimbatm <http://zimbatm.com/> & Hauleth <lukasz@niemier.pl>
+" Version:      0.3
+
+scriptencoding utf-8
+
+let s:direnv_cmd = get(g:, 'direnv_cmd', 'direnv')
+let s:direnv_auto = get(g:, 'direnv_auto', 1)
+let s:job_status = { 'running': 0, 'stdout': [''], 'stderr': [''] }
+
+function! direnv#auto() abort
+  return s:direnv_auto
+endfunction
+
+function! direnv#on_stdout(_, data, ...) abort
+  let s:job_status.stdout = a:data
+endfunction
+
+function! direnv#on_stderr(_, data, ...) abort
+  let s:job_status.stderr = a:data
+endfunction
+
+function! direnv#on_exit(_, status, ...) abort
+  let s:job_status.running = 0
+
+  for l:m in s:job_status.stderr
+    if l:m isnot# ''
+      echom l:m
+    endif
+  endfor
+  exec join(s:job_status.stdout, "\n")
+endfunction
+
+function! direnv#err_cb(_, data) abort
+  call direnv#on_stderr(0, split(a:data, "\n", 1))
+endfunction
+
+function! direnv#out_cb(_, data) abort
+  call direnv#on_stdout(0, split(a:data, "\n", 1))
+endfunction
+
+function! direnv#exit_cb(_, status) abort
+  call direnv#on_exit(0, a:status)
+endfunction
+
+if has('nvim')
+  let s:job =
+        \ {
+        \   'on_stdout': 'direnv#on_stdout',
+        \   'on_stderr': 'direnv#on_stderr',
+        \   'on_exit': 'direnv#on_exit'
+        \ }
+else
+  let s:job =
+        \ {
+        \   'out_cb': 'direnv#out_cb',
+        \   'err_cb': 'direnv#err_cb',
+        \   'exit_cb': 'direnv#exit_cb'
+        \ }
+endif
+
+function! direnv#export() abort
```

Comment:
Splitting this function to `autoload` folder makes no sense as it is launched on `VimEnter`.

Reply:
It is launched on VimEnter, but it requires access to `s:job_status` and `s:direnv_cmd`, which it won’t have if it remains in `plugin/direnv.vim`. It wouldn’t be hard to rewrite this to just use `s:` functions, but given that `direnv#export()` is available, it is *typical* in the VimL that I’ve seen to put functions defined that way in `autoload/`.